chartfx-dataset/src/main/java/de/gsi/dataset/serializer/spi/BinarySerialiser.java

```diff
@@ -95,174 +109,194 @@
         }
-        readBuffer.getString(); // should read "#file producer : "
-        // -- but not explicitly checked
-        final String producer = readBuffer.getString();
-        readBuffer.getString(); // not explicitly checked
-        final byte major = readBuffer.getByte();
-        final byte minor = readBuffer.getByte();
-        final byte micro = readBuffer.getByte();
-
-        final HeaderInfo header = SELF.new HeaderInfo(headerStartField, producer, major, minor, micro);
 
         if (!header.isCompatible()) {
-            final String msg = String.format("byte buffer version incompatible: received '%s' vs. this '%s'", header.toString(), THIS_HEADER.toString());
-            throw new IllegalStateException(msg);
+            final String thisHeader = String.format(" serialiser: %s-v%d.%d.%d", BinarySerialiser.class.getCanonicalName(), VERSION_MAJOR, VERSION_MINOR, VERSION_MICRO);
+            throw new IllegalStateException("byte buffer version incompatible: received '" + header.toString() + "' vs. should '" + thisHeader + "'");
         }
+        buffer.setCallBackFunction(callBackFunction);
         return header;
     }
 
-    public static int[] getArrayDimensions(final IoBuffer readBuffer) {
-        final int arrayDims = readBuffer.getInt(); // array dimensions
-        final int[] dims = new int[arrayDims];
-        for (int i = 0; i < arrayDims; ++i) {
-            dims[i] = readBuffer.getInt();
-        }
-        return dims;
-    }
-
-    public static boolean getBoolean(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getBoolean();
-        }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
-
-    public static boolean[] getBooleanArray(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getBooleanArray();
-        }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
+    public IoBuffer getBuffer() {
+        return buffer;
     }
 
-    public static int getBufferIncrements() {
+    public int getBufferIncrements() {
         return bufferIncrements;
     }
 
-    public static byte getByte(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getByte();
-        }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
-
-    public static byte[] getByteArray(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getByteArray();
-        }
-        throw new IndexOutOfBoundsException(READ_POSITION_AT_BUFFER_END);
-    }
-
-    public static char getCharacter(final IoBuffer readBuffer) {
-        if (readBuffer.hasRemaining()) {
-            return readBuffer.getChar();
+    @Override
+    public <E> Collection<E> getCollection(final Collection<E> collection, final BiFunction<Type, Type[], FieldSerialiser<E>> serialiserLookup) {
```

Comment:
Codacy found an issue: [The method 'getCollection(Collection, BiFunction)' has an NPath complexity of 204, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)